centinel/experiments/baseline_linear.py

```diff
@@ -108,10 +109,18 @@ def run_file(self, input_file):
                     file_comments.append(row)
                 continue
 
+            # detect the header row and store it
+            # it is usually the first row and starts with "url,"
+            if row[0].strip().lower() == "url":
```

Comment:
if you use csv.DictReader and don't specify the field names, it will use the first row as the names of the fields (the keys into the dictionary for each row). You could also make this code more concise by just calling file_contents.readline() before creating the csv reader.